pkg/endpoint/bpf.go

```diff
@@ -0,0 +1,268 @@
+//
+// Copyright 2016 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package endpoint
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"github.com/cilium/cilium/bpf/ctmap"
+	"github.com/cilium/cilium/bpf/geneve"
+	"github.com/cilium/cilium/bpf/policymap"
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/u8proto"
+)
+
+const (
+	ProxyMaxElements = 1024
+)
+
+func (e *Endpoint) writeL4Map(fw *bufio.Writer, owner Owner, m policy.L4PolicyMap, config string) error {
+	array := ""
+
+	for _, l4 := range m {
+		// Represents struct l4_allow in bpf/lib/l4.h
+		protoNum, err := u8proto.ParseProtocol(l4.Protocol)
+		if err != nil {
+			return fmt.Errorf("invalid protocol %s", l4.Protocol)
+		}
+
+		dport := common.Swab16(uint16(l4.Port))
+		redirect := uint16(0)
+
+		redirect = common.Swab16(redirect)
+		entry := fmt.Sprintf("{%d,%d,%d}", dport, redirect, protoNum)
+		if array != "" {
```

Comment:
You can use a slice of strings and use the append
`array = append(array, entry...)` and in the after the loop `prettyString := strings.Join(array, ",")`